File: src/SSHServer/MACAlgorithms/IMACAlgorithm.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SSHServer.MACAlgorithms
{
    public interface IMACAlgorithm : IAlgorithm
    {
        uint KeySize { get; }
        uint DigestLength { get; }
        void SetKey(byte[] key);
        byte[] ComputeHash(uint packetNumber, byte[] data);
    }
}

File: SSHServerNodeJS/SSHServerNodeJS/MACAlgorithms/IMACAlgorithm.ts
import { IAlgorithm } from "../IAlgorithm";

export interface IMACAlgorithm extends IAlgorithm {
    getKeySize(): number;
    getDigestLength(): number;
    setKey(key: Buffer): void;
    computeHash(packetNumber: number, data: Buffer): Buffer;
}
